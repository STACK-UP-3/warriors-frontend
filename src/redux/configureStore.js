import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
