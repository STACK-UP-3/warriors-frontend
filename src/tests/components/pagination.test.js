import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Pagination from '../../components/pagination';
import configureStore from '../../redux/configureStore';

describe('*************** Testing the Pagination  component ***************',()=>{
    const store = configureStore();

    it('Should render the Pagination page correctly', ()=>{
        const wrapper = mount( 
            <Provider store={ store }> 
                <Pagination 
                    previousPage={jest.fn()} 
                    nextPage={jest.fn()} 
                    prevPage={1} 
                    page={ 2 } 
                    numberOfPages={3} 
                    pages={[1,2,3]}
                 /> 
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
