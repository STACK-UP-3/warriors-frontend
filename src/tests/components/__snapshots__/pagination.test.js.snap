// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the Pagination  component *************** Should render the Pagination page correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <_default
    nextPage={[MockFunction]}
    numberOfPages={3}
    page={2}
    pages={
      Array [
        1,
        2,
        3,
      ]
    }
    prevPage={1}
    previousPage={[MockFunction]}
  >
    <div
      className="paginationComponent"
    >
      <ul
        className="pagination"
      >
        <li
          className="disabled"
        >
          <a
            href="#!"
          >
            <i
              className="material-icons"
              id="prevPage"
              onClick={[MockFunction]}
            >
              chevron_left
            </i>
          </a>
        </li>
        <li
          className="active"
        >
          <a
            href="#!"
          >
            2
          </a>
        </li>
        <li>
          <a
            href="#!"
          >
             of 
          </a>
        </li>
        <li>
          <a
            href="#!"
          >
            3
          </a>
        </li>
        <li
          className="waves-effect"
        >
          <a
            href="#!"
          >
            <i
              className="material-icons"
              id="nextPage"
              onClick={[MockFunction]}
            >
              chevron_right
            </i>
          </a>
        </li>
      </ul>
    </div>
  </_default>
</Provider>
`;
